src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -360,6 +362,182 @@ bool emberAfOperationalCredentialsClusterAddOpCertCallback(chip::app::Command *
     return true;
 }
 
+bool emberAfOperationalCredentialsClusterCertChainRequestCallback(chip::app::Command * commandObj, uint16_t certChainType)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    CHIP_ERROR err       = CHIP_NO_ERROR;
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: commissioner has requested Device Attestation Credentials");
+
+    app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID,
+                                         ZCL_CERT_CHAIN_RESPONSE_COMMAND_ID, (chip::app::CommandPathFlags::kEndpointIdValid) };
+
+    TLV::TLVWriter * writer = nullptr;
+
+    VerifyOrExit(commandObj != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    SuccessOrExit(err = commandObj->PrepareCommand(cmdParams));
+    writer = commandObj->GetCommandDataElementTLVWriter();
+    if (certChainType == 1)
+    {
+        SuccessOrExit(err = writer->Put(TLV::ContextTag(0), ByteSpan(dac_certificate, sizeof(dac_certificate))));
+    }
+    else if (certChainType == 2)
+    {
+        SuccessOrExit(err = writer->Put(TLV::ContextTag(0), ByteSpan(pai_certificate, sizeof(pai_certificate))));
+    }
+    else
+    {
+        SuccessOrExit(err = CHIP_ERROR_INTERNAL);
+    }
+    SuccessOrExit(err = commandObj->FinishCommand());
+
+exit:
+    if (status == EMBER_ZCL_STATUS_FAILURE)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed CertChainRequest.");
+        emberAfSendImmediateDefaultResponse(status);
+    }
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to encode response command: %s", ErrorStr(err));
+    }
+
+    return true;
+}
+
+bool emberAfOperationalCredentialsClusterAttestationRequestCallback(chip::app::Command * commandObj,
+                                                                    chip::ByteSpan attestationNonce)
+{
+    EmberAfStatus status    = EMBER_ZCL_STATUS_SUCCESS;
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    TLV::TLVWriter * writer = nullptr;
+    System::PacketBufferTLVWriter tlvWriter;
+    TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
+    System::PacketBufferHandle certDeclaration;
+    System::PacketBufferHandle certDeclarationTbs;
+    System::PacketBufferHandle firmwareInfo;
+    System::PacketBufferHandle attestationElements;
+    System::PacketBufferHandle attestationElementsTbs;
```

Comment:
Please do not use `System::PacketBufferHandle` as heap, but rather employ `ScopedMemoryBuffer`

Reply:
Updated